Sources/PackageLoading/PackageDescription4Loader.swift

```diff
@@ -362,13 +363,19 @@ extension PackageDependencyDescription {
             self = .local(identity: identity,
                           name: name,
                           path: path,
-                           productFilter: .everything)
+                          productFilter: .everything)
         // a package in a git location, may be a remote URL or on disk
         // TODO: consider refining the behavior + validation when the package is on disk
         // TODO: refactor this when adding registry support
         default:
             // location mapping (aka mirrors)
             let location = identityResolver.resolveLocation(from: location)
+            
+            if let validPath = try? AbsolutePath(validating: location), fileSystem.exists(validPath) {
+                let gitRepoProvider = GitRepositoryProvider()
+                try gitRepoProvider.isInitializedDirectory(location: location)
```

Comment:
here is where we should throw the more specific error:

if !gitRepoProvider.isInitializedDirectory(location) {
 throw StringError("Cannot clone from local directory \(location)\nPlease git init or use \"path:\" for \(location)")
}